### PR TITLE
Refactored to improve readability

### PR DESCRIPTION
**Pull request by yashashreesuresh**
This commit refactors the code to improve readability.

### CODE REVIEW

**Comment by yashashreesuresh** [comment]
Please have a look @chaitanyaenr

**Comment by paigerube14** [comment]
LGTM 

This might not be the pr to do it in, but are we able to combine some of the steps with there corresponding logging lines? 
For example, line 143 (if watch_nodes) prints out the iteration node status and line 207 (if not watch_nodes_status) prints out only if there's a failure. Seems like we could just combine and print out only the failure case. I know we had kept them separate so we would be able to print out all the failures at the end but I just think the one loop is getting a little large. Thoughts? @mffiedler @yashashreesuresh @chaitanyaenr

File: cerberus/prometheus/client.py [hunk 1]
```diff
@@ -23,3 +24,27 @@ def get_metrics(query):
         return prom_cli.custom_query(query=query, params=None)
     except Exception as e:
         logging.error("Failed to get the metrics: %s" % e)
+
+
+# Alert on high latencies
+def display_metrics(distribution, prometheus_url, prometheus_bearer_token):
```
Comment (chaitanyaenr) [inline cerberus/prometheus/client.py:30]: Maybe make this a generic function which accepts a prometheus query, runs it and returns the output?  This way we can use it to pass the query of interest ( ALERTS{alertname="KubeAPILatencyHigh", severity="warning"} in this case ) and warn the users with a custom message.

**Comment by chaitanyaenr** [comment]
> LGTM
> 
> This might not be the pr to do it in, but are we able to combine some of the steps with there corresponding logging lines?
> For example, line 143 (if watch_nodes) prints out the iteration node status and line 207 (if not watch_nodes_status) prints out only if there's a failure. Seems like we could just combine and print out only the failure case. I know we had kept them separate so we would be able to print out all the failures at the end but I just think the one loop is getting a little large. Thoughts? @mffiedler @yashashreesuresh @chaitanyaenr

@paigerube14 maybe we can log the namespaces being monitored when Cerberus starts for users to know what namespaces are being monitored and then just log when there's a failure to cut down the noise. Thoughts?

**Comment by paigerube14** [comment]
> > LGTM
> > This might not be the pr to do it in, but are we able to combine some of the steps with there corresponding logging lines?
> > For example, line 143 (if watch_nodes) prints out the iteration node status and line 207 (if not watch_nodes_status) prints out only if there's a failure. Seems like we could just combine and print out only the failure case. I know we had kept them separate so we would be able to print out all the failures at the end but I just think the one loop is getting a little large. Thoughts? @mffiedler @yashashreesuresh @chaitanyaenr
> 
> @paigerube14 maybe we can log the namespaces being monitored when Cerberus starts for users to know what namespaces are being monitored and then just log when there's a failure to cut down the noise. Thoughts?

I think that would work. I can work on this and see what I can trim down if you want? Should I open a new issue to work on or just want to do it in this one?

**Comment by chaitanyaenr** [comment]
> > > LGTM
> > > This might not be the pr to do it in, but are we able to combine some of the steps with there corresponding logging lines?
> > > For example, line 143 (if watch_nodes) prints out the iteration node status and line 207 (if not watch_nodes_status) prints out only if there's a failure. Seems like we could just combine and print out only the failure case. I know we had kept them separate so we would be able to print out all the failures at the end but I just think the one loop is getting a little large. Thoughts? @mffiedler @yashashreesuresh @chaitanyaenr
> > 
> > 
> > @paigerube14 maybe we can log the namespaces being monitored when Cerberus starts for users to know what namespaces are being monitored and then just log when there's a failure to cut down the noise. Thoughts?
> 
> I think that would work. I can work on this and see what I can trim down if you want? Should I open a new issue to work on or just want to do it in this one?

@paigerube14 opening a new issue sounds good, feel free to assign it to yourself. Thanks.

File: cerberus/prometheus/client.py (hunk 1 above)
Comment (yashashreesuresh) [reply cerberus/prometheus/client.py]: Fixed it.

**Comment by yashashreesuresh** [comment]
> LGTM
> 
> This might not be the pr to do it in, but are we able to combine some of the steps with there corresponding logging lines?
> For example, line 143 (if watch_nodes) prints out the iteration node status and line 207 (if not watch_nodes_status) prints out only if there's a failure. Seems like we could just combine and print out only the failure case. I know we had kept them separate so we would be able to print out all the failures at the end but I just think the one loop is getting a little large. Thoughts? @mffiedler @yashashreesuresh @chaitanyaenr

With [multiprocessing](https://github.com/openshift-scale/cerberus/pull/60/files#diff-1804ee9fca6e3735b37749eda21c5aceR195), all the checks are performed in client.py and the loop size significantly decreases once the PR is merged. And we log and store the failures in database in the main loop (https://github.com/openshift-scale/cerberus/blob/master/start_cerberus.py#L245). Therefore, I think we need two different loops.

**Comment by paigerube14** [comment]
> > LGTM
> > This might not be the pr to do it in, but are we able to combine some of the steps with there corresponding logging lines?
> > For example, line 143 (if watch_nodes) prints out the iteration node status and line 207 (if not watch_nodes_status) prints out only if there's a failure. Seems like we could just combine and print out only the failure case. I know we had kept them separate so we would be able to print out all the failures at the end but I just think the one loop is getting a little large. Thoughts? @mffiedler @yashashreesuresh @chaitanyaenr
> 
> With [multiprocessing](https://github.com/openshift-scale/cerberus/pull/60/files#diff-1804ee9fca6e3735b37749eda21c5aceR195), all the checks are performed in client.py and the loop size significantly decreases once the PR is merged. And we log and store the failures in database in the main loop (https://github.com/openshift-scale/cerberus/blob/master/start_cerberus.py#L245). Therefore, I think we need two different loops.

Ah thanks for pointing that out! I missed all that being taken out! I'll retake a deeper look at that PR

**Comment by yashashreesuresh** [comment]
> > > LGTM
> > > This might not be the pr to do it in, but are we able to combine some of the steps with there corresponding logging lines?
> > > For example, line 143 (if watch_nodes) prints out the iteration node status and line 207 (if not watch_nodes_status) prints out only if there's a failure. Seems like we could just combine and print out only the failure case. I know we had kept them separate so we would be able to print out all the failures at the end but I just think the one loop is getting a little large. Thoughts? @mffiedler @yashashreesuresh @chaitanyaenr
> > 
> > 
> > With [multiprocessing](https://github.com/openshift-scale/cerberus/pull/60/files#diff-1804ee9fca6e3735b37749eda21c5aceR195), all the checks are performed in client.py and the loop size significantly decreases once the PR is merged. And we log and store the failures in database in the main loop (https://github.com/openshift-scale/cerberus/blob/master/start_cerberus.py#L245). Therefore, I think we need two different loops.
> 
> Ah thanks for pointing that out! I missed all that being taken out! I'll retake a deeper look at that PR

Yeah, this PR is 11 commits behind the master. We can look at enhancing the readability once the multiprocessing PR gets merged as it would give a better picture.

File: cerberus/prometheus/client.py [hunk 2]
```diff
@@ -23,3 +24,25 @@ def get_metrics(query):
...
+        url = runcommand.invoke(r"""oc get routes -n openshift-monitoring -o=jsonpath='{.items[?(@.metadata.name=="prometheus-k8s")].spec.host}'""") # noqa
+        prometheus_url = "https://" + url
+    if distribution == "openshift" and not prometheus_bearer_token:
+        prometheus_bearer_token = runcommand.invoke("oc -n openshift-monitoring "
+                                                    "sa get-token prometheus-k8s")
+    if prometheus_url and prometheus_bearer_token:
+        initialize_prom_client(prometheus_url, prometheus_bearer_token)
+        metrics = get_metrics(query)
+        if metrics:
+            logging.warning("Kubernetes API server latency is high. "
```
Comment (chaitanyaenr) [inline cerberus/prometheus/client.py:42]: We might want to move this log to start_cerberus.py script as it's specific to the query.

Comment (yashashreesuresh) [reply cerberus/prometheus/client.py]: Fixed it.

**Pull request merged**